### ISSUE TITLE
False positive 'nonexistent-operator' for repeated '-' (with parens)

### ISSUE THREAD

**Issue by skirpichev** [PyCQA/pylint#5769]
### Bug description
```
$ cat -n a.py 
     1  a = 1
     2  b = -(-a)
```

### Configuration

```ini
none
```

### Command used

```shell
pylint -d all -e nonexistent-operator a.py
```

### Pylint output

```shell
************* Module a
a.py:2:4: E0107: Use of the non-existent -- operator (nonexistent-operator)

-----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -30.00/10, +15.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.0+ (heads/3.10:045f205ba4, Nov  8 2021, 05:21:54) [GCC 10.2.1 20210110]
```
